refbox/src/refBox.js

```diff
@@ -176,34 +177,20 @@ class RefBox extends React.Component {
       attempt: '',
       scoreTable: [],
     }
+    this.onMessage = this.onMessage.bind(this);
   }
 
-  // Websocket use based on 
-  // https://dev.to/finallynero/using-websockets-in-react-4fkp
-  // instance of websocket connection as a class property
-  ws = new WebSocket('ws://localhost:6789')
-
-  componentDidMount() {
-    this.ws.onopen = () => {
-      // on connecting, do nothing but log it to the console
-      console.log('connected')
-    }
-
-    this.ws.onmessage = evt => {
-      // listen to data sent from the websocket server
-      const message = JSON.parse(evt.data)
-      console.log('Message: ', message);
-      this.setState({dataFromServer: message})
-      if (this.state.arena in message) {
-        this.updateArenaData(message[this.state.arena]);
-      }
-    }
-
-    this.ws.onclose = () => {
-      console.log('disconnected')
-      // automatically try to reconnect on connection loss
+  onMessage(evt) {
+    // listen to data sent from the websocket server
+    const message = JSON.parse(evt);
+    console.log('Message: ', message);
+    this.setState({dataFromServer: message});
+    if (this.state.arena in message) {
+      this.updateArenaData(message[this.state.arena]);
     }
+  }
 
+  componentDidMount() {
```

Comment:
This is deprecated isn't it?

Reply:
Yes, I don't know react requires this function to be defined.

Reply:
Also works without, removed